frontend/src/context/notificationStateSlice.ts

```diff
@@ -37,12 +39,19 @@ export const notificationStateSlice = createSlice({
   initialState,
   reducers: {
     addNotification: (
-      { notifications, ...rest },
+      state,
       { payload }: PayloadAction<NotificationConstructor>,
-    ) => ({
-      ...rest,
-      notifications: notifications.concat(new Notification(payload)),
-    }),
+    ) => {
+      const { notifications, ...rest } = state;
+      const notification = new Notification(payload);
+      if (notifications.find(n => n.key === notification.key)) {
+        return state;
+      }
+      return {
+        ...rest,
+        notifications: notifications.concat(notification),
```

Comment:
could notifications be changed to a set instead? so that you don't need to do the verification above?

Reply:
Either a set or a map could work. I think the notifications are not used it too many places so if this works we should be fine